### PR TITLE
Fix site extension version check across multiple feeds

### PR DESCRIPTION
**Pull request by EricSten-MSFT**
Tested that leaving an empty version will allow finding the most recent version of a site extension across multiple feeds.

### CODE REVIEW

**Comment by davidebbo** [comment]
Looks like you're still based on the old tree, which is why it shows all the old commit. You need to rebase your work on the latest tree to reduce changes of conflict.

File: Kudu.Core/SiteExtensions/SiteExtensionManager.cs [hunk 1]
```diff
@@ -622,19 +625,28 @@ private static void EnsureInstallationEnviroment(string installationDir, ITracer
...
                 {
                     // case 3
                     UIPackageMetadata remotePackage = await rr.GetLatestPackageByIdFromSrcRepo(id);
                     if (remotePackage != null)
                     {
-                        isInstalled = remotePackage.Identity.Version.ToNormalizedString().Equals(localPackageVersion, StringComparison.OrdinalIgnoreCase) && string.Equals(localPackageInstallationArgs, installationArgs);
+                        tracer.Trace("Found version: {0} on feed: {1}",
+                                     remotePackage.Identity.Version.ToNormalizedString(),
+                                     rr.PackageSource.Source);
+                        if (lastFoundVer < remotePackage.Identity.Version)
```
Comment (davidebbo) [inline Kudu.Core/SiteExtensions/SiteExtensionManager.cs]: Is this check ok if `lastFoundVer` is null?

File: Kudu.Services/Kudu.Services.csproj [hunk 2]
```diff
@@ -11,11 +11,6 @@
     <DocumentationFile>bin\Debug\Kudu.Services.XML</DocumentationFile>
     <TargetFrameworkVersion>v4.6</TargetFrameworkVersion>
     <TargetFrameworkProfile />
-  </PropertyGroup>
-  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
-    <NoWarn>1591</NoWarn>
-  </PropertyGroup>
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Release|AnyCPU'">
```
Comment (davidebbo) [inline Kudu.Services/Kudu.Services.csproj]: That looks like a random diff that probably should not be included.

File: Kudu.SiteManagement/Kudu.SiteManagement.csproj [hunk 3]
```diff
@@ -10,10 +10,10 @@
     <TargetFrameworkVersion>v4.6</TargetFrameworkVersion>
     <TargetFrameworkProfile />
   </PropertyGroup>
-  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
+  <PropertyGroup Condition=" '$(Configuration)' == 'Debug' ">
     <DefineConstants>TRACE;DEBUG;CODE_ANALYSIS;SITEMANAGEMENT</DefineConstants>
   </PropertyGroup>
-  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
+  <PropertyGroup Condition=" '$(Configuration)' == 'Release' ">
```
Comment (davidebbo) [inline Kudu.SiteManagement/Kudu.SiteManagement.csproj]: This file should not be part of it either.

File: Kudu.Services/Kudu.Services.csproj (hunk 2 above)
Comment (EricSten-MSFT) [reply Kudu.Services/Kudu.Services.csproj]: Uh, how do I pull the two files out of the change list?

Comment (davidebbo) [reply Kudu.Services/Kudu.Services.csproj]: Several ways, but one is to just push another commit that restores them. It all gets squashed in the end anyway.

**Pull request merged**